Update ESACCI-SST cmorizer to v3.0

LisaBock opened the pull request:
## Description

Update ESACCI-SST cmorizer to v3.0 (daily and monthly data).
I changed the ESACCI-SST data to the variable `tos` instead of `ts` because the observation field is only defined over ocean and not the whole globe like `ts`. 

Registration at CEDA is needed for downloading the data.

Also a new custom variable `tosStderr` is needed (see [#2470](https://github.com/ESMValGroup/ESMValCore/pull/2470)). It is already merge in the main branch of ESMValCore but not included in the last release.

- Closes #3705
- Link to documentation:

* * *

## Before you get started

<!--
    Please discuss your idea with the development team before getting started,
    to avoid disappointment or unnecessary work later. The way to do this is
    to open a new issue on GitHub.
-->

- [x] [☝ Create an issue](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#contributing-code-and-documentation) to discuss what you are going to do

## Checklist

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

<!-- The next two lines turn the 🛠 and 🧪 below into hyperlinks -->
[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-title)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#code-quality)
- [x] [🛠][1] [Documentation](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#documentation) is available
- [x] [🛠][1] [Tests](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#tests) run successfully
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#list-of-authors) is up to date
- [ ] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-checks) were successful

### [New or updated data reformatting script](https://docs.esmvaltool.org/en/latest/develop/dataset.html)

- [x] [🛠][1] [Documentation](https://docs.esmvaltool.org/en/latest/community/dataset.html#dataset-documentation) is available
- [x] [🛠][1] The dataset has been [added to the CMOR check recipe](https://docs.esmvaltool.org/en/latest/community/dataset.html#testing)
- [ ] [🛠][1] The dataset has been added to the shared [data pools](https://docs.esmvaltool.org/en/latest/community/dataset.html#cmorized-data) of DKRZ and Jasmin by the @ESMValGroup/OBS-maintainers team
- [x] [🧪][2] Numbers and units of the data look [physically meaningful](https://docs.esmvaltool.org/en/latest/community/dataset.html#scientific-sanity-check)

axel-lauer reviewed (changes requested):
This looks already very good. I addition to my comments above, I would recommend the following changes:

- Units are reported as `degC` but should be in `K` (fields also look like data are in `K`)
- The variable attributes `host`, `reference`, `source`, `tier`, `user` and `version` should rather be global attributes
- Not sure about the variable attributes `mip`, `modeling_realm` and `project_id`. Other CMORrized datasets do not have these attributes. I would suggest to either change these attributes to global attributes or to remove them.
- I would remove the variable attribute `comment: Note that the variable tsStderr is an uncertainty not a standard error.` from variable `tos` as this only applies to `tosStderr`, which is written to separate files
- the attribute `long_name` of variable `time` should be simply `time` (instead of `reference time of sst field`)
- similarly, the attribute `long_name` of variables `lon` and `lat` should be `longitude` and `latitude` instead of `longitude coordinate` and `latitude coordinate`

LisaBock commented on esmvaltool/recipes/examples/recipe_check_obs.yml:
```diff
@@ -311,12 +311,13 @@ diagnostics:
+      tos:
+      tosStderr:
     additional_datasets:
-      - {dataset: ESACCI-SST, project: OBS, mip: Amon, tier: 2,
-         type: sat, version: 2.2,
-         start_year: 1982, end_year: 2019}
+      - {dataset: ESACCI-SST, project: OBS6, mip: Omon, tier: 3,
+         type: sat, version: 3.0-L4-analysis, timerange: '200401/200412'}
+      - {dataset: ESACCI-SST, project: OBS6, mip: Oday, tier: 3,
+         type: sat, version: 3.0-L4-analysis, timerange: '200401/200412'}
```
Thanks.

LisaBock commented:
Thanks @axel-lauer for the review! The memory issue should be solved now and also the years could be also set now in the command line. Below you find my answers to your comments:

>     * Units are reported as `degC` but should be in `K` (fields also look like data are in `K`)
Cmor units are degC, the fields is now also converted.

>     * The variable attributes `host`, `reference`, `source`, `tier`, `user` and `version` should rather be global attributes
This happened due to a change in Iris. I updated the function `set_global_atts` in PR #3717 

>     * Not sure about the variable attributes `mip`, `modeling_realm` and `project_id`. Other CMORrized datasets do not have these attributes. I would suggest to either change these attributes to global attributes or to remove them.
They are also global attributes now.

>     * I would remove the variable attribute `comment: Note that the variable tsStderr is an uncertainty not a standard error.` from variable `tos` as this only applies to `tosStderr`, which is written to separate files
The comment is now only attached to 'tosStderr'.

>     * the attribute `long_name` of variable `time` should be simply `time` (instead of `reference time of sst field`)
Thanks. Done.

>     * similarly, the attribute `long_name` of variables `lon` and `lat` should be `longitude` and `latitude` instead of `longitude coordinate` and `latitude coordinate`
Thanks. Done.

axel-lauer reviewed (commented):
Thanks @LisaBock for addressing all my comments. Everything looks fine to me now. The only question mark left is whether the uncertainty data `tsStderr` should really be regridded. Strictly speaking, we would need to know the correlation lengths in order to be able to regrid the data, which we do not have at the moment. One option could be to simply not regrid `tsStderr` so we can look into this at a later stage.

LisaBock commented:
> Thanks @LisaBock for addressing all my comments. Everything looks fine to me now. The only question mark left is whether the uncertainty data `tsStderr` should really be regridded. Strictly speaking, we would need to know the correlation lengths in order to be able to regrid the data, which we do not have at the moment. One option could be to simply not regrid `tsStderr` so we can look into this at a later stage.

Thanks again @axel-lauer !
I removed the regridding of the uncertainty field. As the files are now much huger we might not to cmorize the whole time period for tosStderr...

oembury commented on esmvaltool/cmorizers/data/cmor_config/ESACCI-SST.yml:
```diff
@@ -1,29 +1,29 @@
+    frequency: day
+    filename: ESACCI-L4_GHRSST-SSTdepth-OSTIA-GLOB_CDR3.0-v02.0-fv01.0.nc
+    start_year: 1980
+    end_year: 2023
```
The files in the CEDA archive are "final"
The CDR covers the 1980-2021 period and is generated under ESA CCI funding as a historical reprocessing.
The ICDR is the ongoing extension (daily updates at ~2 weeks to 1 month latency). It is funded by C3S (2022), then EOCIS/UKMCAS (2023/24).
* Archiving the ICDR (i.e. copy to the CEDA archive) is not yet automated, but when implemented we'll archive ICDR data at a latency of 2-3 months.

LisaBock commented:
Hi @axel-lauer , the PR is now ready for a final look!

Last changes:
- downloader is changed to wget (no registration and password is needed anymore)
- data is moved back from Tier3 to Tier2

axel-lauer reviewed (approved):
This looks all good now! Thanks @LisaBock ! I ran the formatting script and successfully produced data from 1980-2021. All output looks good. The downloading script also works (again) since changing from ftp to wget. This is ready to be merged.

sloosvel commented:
Hi, the code freeze for version 2.12 will be on Monday 3rd of March. Is this ready to be included? If this is not ready, let me know and I will move it to the next milestone.

LisaBock commented:
> Hi, the code freeze for version 2.12 will be on Monday 3rd of March. Is this ready to be included? If this is not ready, let me know and I will move it to the next milestone.

Hi @sloosvel , this is ready for merging! As soon as the new Core release is used for the tests, I think.
